spec/workers/detect_invariants_daily_check_spec.rb

```diff
@@ -74,29 +74,61 @@
       )
     end
 
-    it 'detects submitted applications with more than the maximum number of course choices' do
-      allow(Sentry).to receive(:capture_exception).with(an_instance_of(described_class::SubmittedApplicationHasMoreThanTheMaxCourseChoices))
+    context 'when continuous application', continuous_applications: true do
+      it 'detects submitted applications with more than the maximum number of course choices' do
+        allow(Sentry).to receive(:capture_exception).with(an_instance_of(described_class::SubmittedApplicationHasMoreThanTheMaxCourseChoices))
 
-      good_application_form = create(:completed_application_form, submitted_application_choices_count: 4)
-      bad_application_form = create(:completed_application_form, submitted_application_choices_count: 3)
-      bad_application_form.application_choices << build_list(:application_choice, 2, status: :offer)
-      ApplicationChoice.all.each { |a| a.update_course_option_and_associated_fields! create(:course_option) }
+        good_application_form = create(:completed_application_form, submitted_application_choices_count: 4)
```

Comment:
Question: Does the invariant check counts the scenario when the user could have 10 submitted applications (6 inactive and 4 awaiting for example)?

Reply:
good q! – will add